File: doc-examples/example-java/src/main/java/example/Producer.java
package example;

import io.micronaut.pulsar.annotation.PulsarProducer;
import io.micronaut.pulsar.annotation.PulsarProducerClient;
import org.apache.pulsar.client.api.MessageId;

import java.util.concurrent.CompletableFuture;

@PulsarProducerClient // <1>
public interface Producer {
    @PulsarProducer(topic = "persistent://public/default/messages-java-docs", producerName = "java-test-producer") // <2>
    CompletableFuture<MessageId> send(String message); // <3>

    @PulsarProducer(topic = "persistent://public/default/messages-java-docs", producerName = "b-java-test-producer")
    void sendBlocking(String message); // <4>
}

File: doc-examples/example-python/src/main/python/example/Producer.py
from abc import ABC, abstractmethod

from java.util.concurrent import CompletableFuture
from micronaut.pulsar.annotation import PulsarProducer, PulsarProducerClient
from org.apache.pulsar.client.api import MessageId


@PulsarProducerClient  # <1>
class Producer(ABC):

    @PulsarProducer(topic="persistent://public/default/messages-python-docs", producerName="python-test-producer")  # <2>
    @abstractmethod
    def send(self, message: str) -> CompletableFuture[MessageId]:  # <3>
        ...

    @PulsarProducer(topic="persistent://public/default/messages-python-docs", producerName="b-python-test-producer")
    @abstractmethod
    def send_blocking(self, message: str) -> None:  # <4>
        ...
